jenkins_pipelines/scripts/kafka/consumer.py

```diff
@@ -0,0 +1,205 @@
+
+import functools
+import json
+import logging
+import os
+import subprocess
+import time
+import dataclasses
+
+import confluent_kafka
+import git
+import requests
+
+import producer
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='[%(asctime)s] %(levelname)s [KAFKA CONSUMER]: %(message)s'
+)
+
+
+class APIClients:
+    authorization_parameters: tuple = ('manager', os.getenv('JENKINS_API_TOKEN'))
+
+    @staticmethod
+    def log_http_requests(request):
+        @functools.wraps(request)
+        def wrapper(self, *args, **kwargs):
+            response = request(self, *args, **kwargs)
+            logging.info(f"{request.__name__.upper()} {args[0]}, STATUS: {response.status_code}")
+            if response.status_code not in (200, 201):
+                logging.error(f"{response.content}")
+            return response
+        return wrapper
+
+    @log_http_requests
+    def get(self, endpoint: str):
+        return requests.get(endpoint, auth=self.authorization_parameters, verify=False, timeout=10)
+
+    @log_http_requests
+    def post(self, endpoint: str, params=None, data=None):
+        if endpoint.startswith('https://hooks.slack.com'):
+            return requests.post(endpoint, headers={'Content-Type': 'application/json'}, data=json.dumps(data), timeout=10)
+        return requests.post(endpoint, auth=self.authorization_parameters, params=params, verify=False, timeout=10)
+
+
+@dataclasses.dataclass
+class KafkaConsumer:
+    consumer = confluent_kafka.Consumer({
+        'bootstrap.servers': 'localhost:9092',
+        'group.id': 'jenkins_pipelines',
+        'auto.offset.reset': 'earliest',
+        'enable.auto.commit': False,
+        'max.poll.interval.ms': 86400000
+    })
+    kafka_topic = 'sle_mu_43'
+    api_clients = APIClients()
+
+    def __post_init__(self) -> None:
+        self.consumer.subscribe([self.kafka_topic])
+
+    @staticmethod
+    def pull_latest_susemanager_ci():
+        susemanager_ci_repository = git.Repo('/home/appuser/susemanager-ci')
+        try:
+            susemanager_ci_repository.remotes.origin.pull()
+        except git.exc.GitError as stderr:
+            logging.warning(f"Error during git pull on susemanager-ci repository: {stderr}")
+
+    @staticmethod
+    def generate_custom_repositories(incidents: dict):
+        incident_numbers = ','.join(
+            str(incident['incident']['incident_id'])
+            for incident in incidents['data']
+        )
+        subprocess.run(
```

Comment:
Having set `check=True`, if the JSON generating scripts fails, for whatever reason, this function call will raise a `subprocess.CalledProcessError`

Maybe we should handle the possibility of a failure here, or where the function is called, and at least log some useful info.

Reply:
Good catch, done